app/models/draw.rb

```diff
@@ -122,6 +123,24 @@ def lottery_complete?
     groups.all? { |g| g.lottery_number.present? }
   end
 
+  # Query method to check whether or not a draw is oversubscribed by checking
+  # all groups. Memoized to avoid querying multiple times per request.
+  #
+  # @return [Boolean] whether or not the draw is oversubscribed
+  def oversubscribed?
+    @oversubscribed ||= GroupSizesQuery.new(groups).call.any? do |size|
+      groups.where(size: size).count > suites.where(size: size).count
```

Comment:
Yea I guess they could be scopes but I'm not sure how much you benefit from that (it's a pretty self-explanatory query). Either way, if you feel strongly open up a refactoring issue.